### ISSUE TITLE
StringLiteralExprSyntax adds whitespace in front of closingQuote

### ISSUE THREAD

**Issue by Jeehut** [swiftlang/swift-syntax#2906]
### Description

I'm using version `600.0.1` and wrote this code as part of a macro expression:

```swift
      let defaultValueLiteral = StringLiteralExprSyntax(
         openingQuote: .stringQuoteToken(),
         segments: StringLiteralSegmentListSyntax([.stringSegment(.init(content: "Some Text"))]),
         closingQuote: .stringQuoteToken()
      )
```

When I return it in my `ExpressionMacro` as in `return ExprSyntax(defaultValueLiteral)`, the expanded code looks like this:

```swift
"Some Text "
```

But I expect it to be:

```swift
"Some Text"
```

### Steps to Reproduce

Here's a more full code example:

```swift
public struct TranslationKey: ExpressionMacro {
   /// Constructing code like: `String(localized: "MyView.Body.Button.saveChanges", defaultValue: "Save Changes")`
   public static func expansion(
      of node: some FreestandingMacroExpansionSyntax,
      in context: some MacroExpansionContext
   ) throws -> ExprSyntax {
      // constructing: `"Some Text"`
      let defaultValueLiteral = StringLiteralExprSyntax(
         openingQuote: .stringQuoteToken(),
         segments: StringLiteralSegmentListSyntax([.stringSegment(.init(content: "Some Text"))]),
         closingQuote: .stringQuoteToken()
      )

      // I have actually more code here (just in case you were wondering)

      return ExprSyntax(defaultValueLiteral)
   }
}
```

**Comment by ahoppen** [comment]
Synced to Apple’s issue tracker as rdar://140829044

**Comment by Jeehut** [comment]
Firstly: The code I posted above is **exactly** the code with which I can reproduce the issue. The comment was just merely to tell you that this is not my final macro and there's more work I need to do. So it can't be related to any `DeclSyntax` stuff I'm doing somewhere else. Did you try this code on your end and didn't run into the same issue?

Secondly: I do not get any log messages when running my tests. I set a breakpoint inside `logStringInterpolationParsingError` at the line you specified (it was line `62` for me) but when running my tests, the breakpoint didn't get hit.

For additional context, here's the test code I've written (using point-free's [`MacroTesting` library](https://github.com/pointfreeco/swift-macro-testing)):

```swift
import Testing
import MacroTesting
import SwiftSyntaxMacrosTestSupport
import TranslateKitMacros
@testable import TranslateKit

@Test
func simpleViewButtonExpansion() {
   assertMacro(["tk": TranslationKey.self]) {
      """
      import SwiftUI
      
      struct MyView: View {
         var body: some View {
            Button(#tk("Save Changes")) { 
               self.handleSave()
            }
         }
      }
      """
   } expansion: {
      """
      import SwiftUI

      struct MyView: View {
         var body: some View {
            Button("Save Changes")) { 
               self.handleSave()
            }
         }
      }
      """
   }
}
```

Note that I tested the macro also in a real-world project and hit the same issue, so I don't think it's related to how the test is written.

I stripped down my Swift package to a minimal reproduction variant and zipped it for you. This way you should be able to fully reproduce my reported issue by simply running the tests:
[TranslateKit-Issue-2906.zip](https://github.com/user-attachments/files/18006554/TranslateKit-Issue-2906.zip)

**Comment by Jeehut** [comment]
@ahoppen Awesome, thank you! That actually fixed the issue I've had. But the whole API is not very intuitive. Is there any good place to learn about all these things, like a book or at least some unit tests which I can read to learn things like this?

**Comment by ahoppen** [comment]
Unfortunately I don’t have any good single resource that I can recommend to learn about these sorts of details.

**Comment by ahoppen** [comment]
A general “Need more documentation” issue will not be very helpful or actionable but I think that more concrete questions that are left unanswered by the documentation at https://swiftpackageindex.com/swiftlang/swift-syntax/main/documentation/swiftsyntax can be helpful. Also: If you have found an answer to one of these questions and would like to share it with the community, PRs are always very welcome and I’m happy to help polish it, if necessary.

**Comment by Jeehut** [comment]
@ahoppen I see, that makes sense. I think, what would have helped me most in this specific situation would have been if each `Decl` had some common usage examples both for reading from and creation of each Declaration. In my particular case, if `StringLiteralExprSyntax` had an example in the type documentation how to create one with a string (and other common ways if there are any), I could have discovered it right within Xcode when playing around with the APIs, cause I often alt-click on types or initializers to understand them better if I don't get what I expected from the API naming itself.

So, you would suggest that I just document what I've learned this way inside the respective types right within the project and create a pull request right away without creating a separate issue for each? I may actually create a bigger PR documenting a whole bunch of APIs with some examples I've run into myself. Cause why not if I'm already at it. Is a "big" documentation-adding PR fine with you or rather small ones documented only few types each time?

**Comment by ahoppen** [comment]
Yes, a PR without an issue is perfectly fine. Thank you in advance for doing it 🙏🏽. As long as the PR doesn’t get massive (I usually fine 1000 LOC a point when things start to get hard to review), one PR is just fine.

And I would attach the documentation for all the types to the type nodes, like here: 

https://github.com/swiftlang/swift-syntax/blob/b479382518694149d8d25d7ede0443f88958993a/CodeGeneration/Sources/SyntaxSupport/DeclNodes.swift#L251-L277

**Comment by Jeehut** [comment]
@ahoppen Here we go (probably done wrong though): https://github.com/swiftlang/swift-syntax/pull/2922
